petastorm/arrow_reader_worker.py

```diff
@@ -157,7 +157,21 @@ def _load_rows(self, pq_file, piece, shuffle_row_drop_range):
         result = self._read_with_shuffle_row_drop(piece, pq_file, column_names, shuffle_row_drop_range)
 
         if self._transform_spec:
-            result = pa.Table.from_pandas(self._transform_spec.func(result.to_pandas()), preserve_index=False)
+            result_as_pandas = result.to_pandas()
+            # A user may omit `func` value if they intend just to delete some fields using the TransformSpec
+            if self._transform_spec.func:
+                transformed_result = self._transform_spec.func(result_as_pandas)
+            else:
+                transformed_result = result_as_pandas
+
+            # If transform function left a field that is listed in transform_spec's remove_fields, we remove it
+            # ourselves. Allows for the following transform-spec objects to be created:
+            # TransformSpec(removed_fields=['some field'])
```

Comment:
Similar comment as in `py_dict_reader_worker` could save few lines here. (I believe same time complexity though)

```python
for field_to_remove  in (transformed_result.columns & transform_spec.removed_fields):
    del row[field_to_remove]
```